crates/language/src/lib.rs

```diff
@@ -36,6 +37,7 @@ impl_lang!(Java, language_java);
 impl_lang!(JavaScript, language_javascript);
 impl_lang!(Kotlin, language_kotlin);
 impl_lang!(Lua, language_lua);
+impl_lang!(Scala, language_scala);
```

Comment:
`impl_lang!` is doing stub work here. If the new language does not special process. We can just use this macro to generate the methods needed to support parsing and also declare a new struct called `Scala`.

Scala seems to support `$VAR` as valid identifier name so we don't need customization for `expando_char`.

On the other hand, other languages like Python does not allow `$` in id so we define a `mod python` and import  the language by importing it with `use python::Python`.

I think it is okay to use Scala as is. WDYT?

Reply:
Thanks for the review and explanation! I've enabled allowing edits by maintainers. So you should be able to push to my branch to make any further changes. Otherwise I'm also fine with this being merged now

Reply:
Let me merge it first! Thank you for your contribution!